Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -739,15 +800,16 @@ ExceptionOr<Ref<SourceBuffer>> MediaSource::addSourceBuffer(const String& type)
     if (type.isEmpty())
         return Exception { ExceptionCode::TypeError };
 
-    // 2. If type contains a MIME type that is not supported ..., then throw a
-    // NotSupportedError exception and abort these steps.
-    Vector<ContentType> mediaContentTypesRequiringHardwareSupport;
-    mediaContentTypesRequiringHardwareSupport.appendVector(settings().mediaContentTypesRequiringHardwareSupport());
-
     auto context = scriptExecutionContext();
     if (!context)
         return Exception { ExceptionCode::NotAllowedError };
 
+    // 2. If type contains a MIME type that is not supported ..., then throw a
+    // NotSupportedError exception and abort these steps.
+    Vector<ContentType> mediaContentTypesRequiringHardwareSupport;
+    if (RefPtr document = dynamicDowncast<Document>(context))
+        mediaContentTypesRequiringHardwareSupport.appendVector(document->settings().mediaContentTypesRequiringHardwareSupport());
+
```

Comment:
Can you eliminate the `document` requirement here by using `scriptExecutionContext()->settingsValues()`?

Reply:
SettingsValues doesn't support Vector.